test/clojure/core_test/partial.cljc

```diff
@@ -0,0 +1,15 @@
+(ns clojure.core-test.partial
+  (:require [clojure.test :as t]))
+
+(defn test-fn [& args]
+  (into [] args))
+
+(t/deftest test-partial []
+  (let [simple-use (partial inc 2)]
+    (t/is (= 3 (simple-use))))
+  (let [lazily-evaluated (partial inc 2 3)]
+    #?(:clj  (t/is (thrown? Exception (lazily-evaluated)))
+       :cljs (t/is (thrown? js/Error  (lazily-evaluated)))))
```

Comment:
I don't know if handling the JS case is needed, but I thought I'd add this. I can remove if that makes more sense.